doc/watermarks_HLD.md

```diff
@@ -0,0 +1,212 @@
+Interface                    PG0   PG1   PG2   PG3   PG4   PG5   PG6   PG7
+Ethernet0                     0     0     0    23     0     0     0     0
+…
+Ethernet128                   0     0     0     0     0     0     0     0
+```
+$ show priority-group [watermark|persistent-watermark] ingress [-p N]
+```
+Ingress shared pool occupancy per PG:
+Interface                   PG0   PG1   PG2   PG3   PG4   PG5   PG6   PG7
+Ethernet0                   0  1092     0   380     0     0     0     0
+…
+Ethernet128                 0     0     0     0     0     0     0     0
+```
+$ show queue [watermark|persistent-watermark] egress unicast [-p N]
+```
+Egress shared pool occupancy per unicast queue:
+Interface                    UC0   UC1   UC2   UC3   UC4   UC5   UC6   UC7
+Ethernet0                      0    14     0    11     0     1     0     0
+…
+Ethernet128                    0     0     0     0     0     0     0     0
+```
+$ show queue [watermark|persistent-watermark] egress multicast [-p N]
+```
+Egress shared pool occupancy per multicast queue:
+Interface                    MC0   MC1   MC2   MC3   MC4   MC5   MC6   MC7
+Ethernet0                      0     3     0     0     0     0     0     0
+…
+Ethernet128                    0     0     0     0     0     0     0     0
+```
+
+In addition clear functionality will be added:
+
+$ clear priority-group [watermark|persistent-watermark] headroom
+
+$ clear priority-group [watermark|persistent-watermark] shared
+
+$ clear queue [watermark|persistent-watermark] unicast
+
+$ clear queue [watermark|persistent-watermark] mutlicast
+
+### 3.1.2 Counter DB
+
+#### The following new Queue counters should be available for each queue entry in the DB:
+- "COUNTERS:queue_vid"
+  - SAI_QUEUE_STAT_SHARED_WATERMARK_BYTES
+#### For every Ingress PG the following should be available in the DB:
+- "COUNTERS:pg_vid"
+  - SAI_INGRESS_PRIORITY_GROUP_STAT_XOFF_ROOM_WATERMARK_BYTES
+  - SAI_INGRESS_PRIORITY_GROUP_STAT_SHARED_WATERMARK_BYTES
+#### Additionally a few mappings should be added:
+- "COUNTERS_PG_PORT_MAP" - map PG oid to port oid
+- "COUNTERS_PG_NAME_MAP" - map PG oid to PG name
+- "COUNTERS_PG_INDEX_MAP" - map PG oid to PG index
+
+#### New tables will be introduced: 
+ - "TELEMETRY_WATERMARKS" - containing the watermarks for the telemetry to consume
+ - "PERSISTENT_WATERMARKS" - containing persistent watermarks, cleared only by user and not by telemetry.
+
+### 3.1.3 Orchestration Agent
+Portorch should be updated:
+ - add SAI_QUEUE_STAT_SHARED_WATERMARK_BYTES to queueStatIds
+ - implement new counter group for PG and priorityGroupStatIds
+ - implement PG to port map generation
+
+New watermark orch should be implemented with the following functionality:
+ - Read TELEMETRY_PERIOD from config DB, start a timer for periodically providing telemetry watermark.
+ - Every telemetry period put current watermark into TELEMETRY_WATERMARK table, than clear it on hardware (with clear stat api).
```

Comment:
Periocially, the flex counter put the current watermark into WATERMARK table, and then clear it on hardware.